src/mcx_core.cu

```diff
@@ -1385,6 +1385,60 @@ __device__ inline int launchnewphoton(MCXpos* p, MCXdir* v, Stokes* s, MCXtime*
                     canfocus = (gcfg->srctype == MCX_SRC_SLIT);
                     break;
                 }
+                case (MCX_SRC_RING): {
+                    /* ring light source, can be a complete ring or a truncated ring. the expected parameters for definition of the
+                    light source are as follows:
```

Comment:
please format this using doxygen formatted comments, like the rest of the code, see example here 
https://www.doxygen.nl/manual/docblocks.html